servers/rendering/renderer_rd/renderer_compositor_rd.h

```diff
@@ -145,7 +145,7 @@ class RendererCompositorRD : public RendererCompositor {
 		low_end = false;
 	}
 
-	static RendererCompositorRD *singleton;
+	static RendererCompositorRD *get_singleton() { return static_cast<RendererCompositorRD *>(singleton); }
```

Comment:
This is nitpicking, but I personally prefer having subclasses have their own singleton member. 

In this case if I have instanced an OpenGL compositor, I can still call `RendererCompositorRD::get_singleton` but now I get the pointer to the OpenGL compositor cast to the Render device compositor.

By having each subclass maintain their own properly typed singleton, we don't need to do casts and we get a nullptr returned if we've instanced a different compositor subclass and somehow run through code meant for render device. It's a few extra bytes but it saves us hard to track crashes.

Reply:
Btw, just to be clear, absolutely agree with the move to using `get_singleton`, just that `static RendererCompositorRD *singleton;` should now be a private property that is set in `RendererCompositorRD::RendererCompositorRD` instead of re-using the singleton set in the parent class.